Sources/_StringProcessing/Engine/Processor.swift

```diff
@@ -59,53 +77,50 @@ struct Processor<
 
   var failureReason: Error? = nil
 
-
   // MARK: Metrics, debugging, etc.
   var cycleCount = 0
   var isTracingEnabled: Bool
-
 }
 
 extension Processor {
   typealias Position = Input.Index
 
-  var start: Position { bounds.lowerBound }
-  var end: Position { bounds.upperBound }
+  var start: Position { subjectBounds.lowerBound }
+  var end: Position { subjectBounds.upperBound }
 }
 
 extension Processor {
   init(
     program: MEProgram<Input>,
     input: Input,
-    bounds: Range<Position>,
+    subjectBounds: Range<Position>,
+    searchBounds: Range<Position>,
     matchMode: MatchMode,
     isTracingEnabled: Bool
   ) {
     self.controller = Controller(pc: 0)
     self.instructions = program.instructions
     self.input = input
-    self.bounds = bounds
+    self.subjectBounds = subjectBounds
+    self.searchBounds = searchBounds
     self.matchMode = matchMode
     self.isTracingEnabled = isTracingEnabled
-    self.currentPosition = bounds.lowerBound
+    self.currentPosition = searchBounds.lowerBound
 
-    self.registers = Registers(program, bounds.upperBound)
+    // Initialize registers with end of search bounds
+    self.registers = Registers(program, searchBounds.upperBound)
     self.storedCaptures = Array(
        repeating: .init(), count: program.registerInfo.captures)
 
     _checkInvariants()
   }
 
-
-  mutating func reset(searchBounds: Range<Position>) {
-    // FIXME: We currently conflate both subject bounds and search bounds
-    // This should just reset search bounds
-    self.bounds = searchBounds
-    self.currentPosition = self.bounds.lowerBound
+  mutating func reset(start: Position, end: Position) {
```

Comment:
Do we need another notion of resetting for the purposes of an engine-baked all matches?

Reply:
I modified `Processor.reset` again to only take the new starting position, since its `searchBounds` stay the same through a single call to `firstMatch`. I think you mentioned elsewhere that we eventually want to use a single processor for all the searches in `"foo".matches(of: regex)` — when we do that we'll need to change this again to allow `searchBounds` to be modified at the start of each successive search.